Paths comparison chart

joethreepwood opened the pull request:
## Changes

It's like the other comparison charts. But for paths. 

Tagging Neil as feature owner to review. 

## Checklist
- [ ] Titles are in [sentence case](https://apastyle.apa.org/style-grammar-guidelines/capitalization/sentence-case)
- [ ] Feature names are in [title case](https://apastyle.apa.org/style-grammar-guidelines/capitalization/title-case)
- [ ] Words are spelled using American English
- [ ] I have checked out our [style guide](https://github.com/PostHog/posthog.com/blob/master/STYLEGUIDE.md)

neilkakkar commented:
I love these, great job!

---


What does repeating steps mean? If it's something like `pageview -> pageview -> pageview` becoming `pageview`, we automatically do that :)

Few more fun things that are possible with paths: (feel free to pick and choose what makes sense for comparison)

1. Show paths for people who dropped off / converted in a funnel (this is wow, and I'd really like it to be included here)
2. Customize paths based on number of people who followed that path (advanced option)
3. Create paths using both start and end points
4. Exclude not just events, but wildcards as well.
5. Not just simple wildcards, but any arbitrary regex to clean paths

And things we don't do which are useful for some: (I think Amplitude does this)

1. Zoom further into a path to see more

joethreepwood commented:
> What does repeating steps mean? If it's something like `pageview -> pageview -> pageview` becoming `pageview`, we automatically do that :)

Ah, this was news to me. Why do we do this automatically?

> other things

Rolled these in, thanks!

Mind taking one more look and merging if you're happy?

neilkakkar commented:
> Why do we do this automatically?

... Poor example from me, it's actually true for custom events, not pageview events where the url changes.

It's to discard noise, say for example: someone is on posthog.com, then they refresh the page, they're still on posthog.com but it's a new pageview event. This seems to clutter the visualisation, while not really giving extra info, so we discard these.

neilkakkar commented:
Q: Was this supposed to be reversed:

<img width="899" alt="image" src="https://user-images.githubusercontent.com/7115141/170696155-09aab406-8336-405c-8d91-e836f7e7eb4c.png">

Also curious what 'define users at each step' means?

joethreepwood commented:
> > Why do we do this automatically?
> 
> ... Poor example from me, it's actually true for custom events, not pageview events where the url changes.
> 
> It's to discard noise, say for example: someone is on posthog.com, then they refresh the page, they're still on posthog.com but it's a new pageview event. This seems to clutter the visualisation, while not really giving extra info, so we discard these.

Ah, actually a poor question from me. What I mean is: Why do this automatically without giving users an option to enable/disable it (as others do). There may be occasions where users _want_ the noise?


> Q: Was this supposed to be reversed:
> Also curious what 'define users at each step' means?

Fixed this. What I'm trying to describe is this

> Customize paths based on number of people who followed that path (advanced option)

But in a way which takes only a few words. I've changed it to match the product more closely ('Define number of users on path') but let me know if you have any other ideas? It's a tricky one.

neilkakkar commented:
> But in a way which takes only a few words. I've changed it to match the product more closely ('Define number of users on path') but let me know if you have any other ideas? It's a tricky one.

Ah indeed. And makes sense 😅 

---

>  There may be occasions where users want the noise?

Fair question, but no one has asked for this yet, so we're defaulting to not having it.

joethreepwood commented:
> Fair question, but no one has asked for this yet, so we're defaulting to not having it.

Makes sense. Happy for me to merge?
